### PR TITLE
refactor(AjaxObservable): remove needles type param T from AjaxObservable.getJSON()

### PR DESCRIPTION
**Pull request by saneyuki**
**Description:**

BREAKING CHANGE: 
- Remove the unused type parameter `T`
  - By this change, the user does not have to specify needless type parameter.

**Related issue (if exists):**

### CODE REVIEW

**Comment by jayphelps** [comment]
Minor change: "BREAKING CHANGE" messages in commits need to be fully descriptive on their own, independent of the first part of the commit message because when the CHANGELOG is generated, they live in their own section.

e.g. https://github.com/ReactiveX/RxJS/commit/aa30af2 generates to https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#breaking-changes-1

So in this case, something _like_

> BREAKING CHANGE: `Observable.ajax.getJSON()` now only supports a single type parameter, `getJSON<R>(url: string, headers?: Object): Observable<R>`. The extra type parameter it accepted previously was superfluous.

Double check all your other PRs have good BREAKING CHANGE messages too please. Also--related refactors like these should prolly have been in a single PR to ease in the discussion and tracking. #2072 #2071 #2070 #2068

**Comment by saneyuki** [comment]
> Minor change: "BREAKING CHANGE" messages in commits need to be fully descriptive on their own, independent of the first part of the commit message because when the CHANGELOG is generated, they live in their own section.

Okay, I'll update.

> related refactors like these should prolly have been in a single PR to ease in the discussion and tracking. #2072 #2071 #2070 #2068

They are originated from the same issues, but I thought they still are different issue. cleaning up codes or changing signature which we may need some decisions.

**Comment by saneyuki** [comment]
updated the commit message.

File: src/observable/dom/AjaxObservable.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export interface AjaxCreationMethod {
   post(url: string, body?: any, headers?: Object): Observable<AjaxResponse>;
   put(url: string, body?: any, headers?: Object): Observable<AjaxResponse>;
   delete(url: string, headers?: Object): Observable<AjaxResponse>;
-  getJSON<T, R>(url: string, headers?: Object): Observable<R>;
+  getJSON<R>(url: string, headers?: Object): Observable<R>;
```
Comment (blesh) [inline src/observable/dom/AjaxObservable.ts]: should probably just be `T`... that's what we do everywhere else.

Comment (saneyuki) [inline src/observable/dom/AjaxObservable.ts]: @blesh okay. I fixed in https://github.com/ReactiveX/rxjs/pull/2069/commits/501bac49bab8949a64f6b76a47ac47ec6b1bfec1

**Pull request merged**